websites/0-9/1Shows/presence.ts

```diff
@@ -0,0 +1,194 @@
+})
+
+const browsingTimestamp = Math.floor(Date.now() / 1000)
+
+enum ActivityAssets {
+  Logo = 'https://i.imgur.com/MCJ61nd.png',
+}
+
+presence.on('UpdateData', async () => {
+  let presenceData: PresenceData = {
+    largeImageKey: ActivityAssets.Logo,
+    startTimestamp: browsingTimestamp,
+    details: 'Unsupported Page',
+  }
+
+  const { pathname } = document.location
+
+  const [privacy] = await Promise.all([presence.getSetting<boolean>('privacy')])
+
+  if (privacy) {
+    presenceData.details = 'Watching 1Shows'
+    presence.setActivity(presenceData)
+    return
+  }
+
+  const pages: Record<string, PresenceData> = {
+    '/': {
+      details: 'Viewing HomePage 🏠',
+      smallImageKey: Assets.Viewing,
+    },
+    '/profile': {
+      details: 'Viewing Profile 👤',
+      smallImageKey: Assets.Viewing,
+    },
+    '/tv': {
+      details: 'Browsing TV Shows 📺',
+      smallImageKey: Assets.Viewing,
+    },
+    '/search': {
+      details: 'Browsing Search 🔎',
+      smallImageKey: Assets.Viewing,
+    },
+  }
+
+  for (const [path, data] of Object.entries(pages)) {
+    if (pathname === path) {
+      presenceData = {
+        ...presenceData,
+        ...data,
+        type: ActivityType.Watching,
+      }
+    }
+  }
+
+  if (pathname.includes('/movies/')) {
+    switch (pathname.replace(/^\/+/, '').split('/')[0]) {
+      case 'movies': {
+        const match = pathname.match(/\/movies\/(\d+)(?:-([^/]+))?/)
+
+        if (match && match[1]) {
+          const movieName = match[2]?.replace(/-/g, ' ') || 'Unknown Movie'
+
+          const formattedMovieName = movieName
+            .split(' ')
+            .map(word => word.charAt(0).toUpperCase() + word.slice(1))
+            .join(' ')
+
+          presenceData.name = `Watching ${formattedMovieName}`
+          presenceData.details = '1Shows.com'
+
+          const ratingElement = document.querySelector('.radial-progress span.text-white')
+          const rating = ratingElement?.textContent?.trim() || 'N/A'
+
+          const runtimeElement = document.querySelector('#Movie\\ Runtime time p')
+
+          const runtime = runtimeElement?.textContent?.match(/\d+/)?.[0] || 'N/A'
+
+          const releaseDateElement = document.querySelector('#Movie\\ Release\\ Date time p')
+
+          let releaseDate = releaseDateElement?.textContent?.trim() || 'N/A'
+
+          if (releaseDate !== 'N/A') {
+            const dateParts = releaseDate.split(', ')
+            if (dateParts.length === 3) {
+              releaseDate = `${dateParts[1]} ${dateParts[2]}`
+            }
+          }
+
+          presenceData.state = `⭐ ${rating} 🕒 ${runtime} mins 🗓️ ${releaseDate}`
+
+          const posterElement = document.querySelector('figure img.object-cover')
+
+          const posterSrc = posterElement?.getAttribute('src')
+
+          presenceData.largeImageKey = posterSrc
+
+          // Check URL parameter for streaming
+          const urlParams = new URLSearchParams(window.location.search)
```

Comment:
We prefer document over window
```suggestion
          const urlParams = new URLSearchParams(document.location.search)
```